### ISSUE TITLE
There should be an option to specify S3 bucket policy

### ISSUE THREAD

**Issue by milosgajdos** [ministryofjustice/bootstrap-cfn#7]
We should be able to specify S3 bucket policy to limit the access to the newly created S3 buckets.

**Comment by kyriakosoikonomakos** [comment]
Ability to add policy json files will be added. If a policy file is not specified the default would be "free for all".

**Comment by milosgajdos** [comment]
To make this clear to other people reading this issue: by default `bootstrap-cfn` assigns `[Get, Put, List]` policies. 

If you want to override these you need to provide a policy json file and reference it in your cfn configuration file like this:

``` yaml
s3:
    static-bucket-name: my-bucket
    policy: /path/to/my/policy/file.json
```

Policy file could look like this:

``` json
{
    "Resource": "arn:aws:s3:::my-bucket/*",
    "Action": [
        "s3:Get*",
        "s3:Put*",
        "s3:List*",
        "s3:Delete*"
    ],
    "Effect": "Allow"
}
```

**Comment by ashb** [comment]
For the security group policy on the EC2 instances we put the rules directly in the project YAML - we should do the same here rather than referencing another file.

``` yaml
bucket_policy:
    "Resource": "arn:aws:s3:::my-bucket/*"
    "Action": 
        - "s3:Get*"
        - "s3:Put*"
        - "s3:List*"
        - "s3:Delete*"
    "Effect": "Allow"
```

**Comment by ashb** [comment]
Kyriakos points out that even simple policies are quite complex, so lets not put it in the YAML

**Comment by ashb** [comment]
Closed by #18
